src/core/dense_set.cc

```diff
@@ -840,4 +840,20 @@ size_t DenseSet::SizeSlow() {
   return size_;
 }
 
+size_t DenseSet::IteratorBase::TraverseApply(DensePtr* ptr, std::function<void(DensePtr*)> fun) {
+  if (!ptr->IsLink()) {
+    fun(ptr);
```

Comment:
I added LOGS to when I pushed this to check TraverseApply and your comment explains why there were links with only one element -- I was missing the last one :/ I fixed this. Also does that mean we have the same issue in `Scan` ?

```
    // Check home bucket
    if (!curr->IsEmpty() && !curr->IsDisplaced()) {
      // scanning add all entries in a given chain
      while (true) {
        cb(curr->GetObject());
        if (!curr->IsLink())
          break;
  
        DensePtr* mcurr = const_cast<DensePtr*>(curr);
  
        if (ExpireIfNeeded(mcurr, &mcurr->AsLink()->next) && !mcurr->IsLink()) {
          break;
        }   
        curr = &curr->AsLink()->next;
      }   
    }
```

`cb` not called on the last element because we `break` when `if (!curr->IsLink())`

Reply:
seems like it! how come our tests did not catch it?

Reply:
I do not know! I will fix it :smile:

Reply:
code is correct :rofl: we apply the `cb` on the last `node` and `GetObject()` takes care of that... :facepalm: